main.tf

```
@@ -1,4 +1,5 @@
 locals {
-  public_enabled  = module.this.enabled && var.type == "public"
-  private_enabled = module.this.enabled && var.type == "private"
+  public_enabled     = module.this.enabled && var.type == "public"
+  private_enabled    = module.this.enabled && var.type == "private"
+  availability_zones = (module.this.enabled) ? var.availability_zones : []
```

The new style guide is 
```suggestion
  enabled = module.this.enabled
  
  public_enabled     = local.enabled && var.type == "public"
  private_enabled    = local.enabled && var.type == "private"
  availability_zones = local.enabled ? var.availability_zones : []
```